design/unnumbered-bgp.md

```diff
@@ -0,0 +1,436 @@
+        neighbor MTLB activate
+    exit-address-family
+    !
+    address-family ipv6 unicast
+        redistribute connected
+        neighbor MTLB activate
+    exit-address-family
+    exit
+```
+
+An example ToR configuration when based on Junos
+
+```
+//TODO
+```
+
+###  How a peering will look like
+
+```
+> docker exec -it unnumbered-p2p-peer vtysh -c 'show bgp summary'
+
+IPv4 Unicast Summary (VRF default):
+BGP router identifier 11.11.11.254, local AS number 65004 vrf-id 0
+BGP table version 4
+RIB entries 3, using 288 bytes of memory
+Peers 3, using 39 KiB of memory
+Peer groups 1, using 64 bytes of memory
+
+Neighbor                  V         AS   MsgRcvd   MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd   PfxSnt Desc
+kind-control-plane(eth00) 4      65000       510       510        4    0    0 00:24:43            1        2 k8s-node
+kind-worker(eth01)        4      65000       510       510        4    0    0 00:24:43            1        2 k8s-node
+kind-worker2(eth02)       4      65000       510       510        4    0    0 00:24:43            1        2 k8s-node
+
+Total number of neighbors 3
+
+IPv6 Unicast Summary (VRF default):
+BGP router identifier 11.11.11.254, local AS number 65004 vrf-id 0
+BGP table version 3
+RIB entries 1, using 96 bytes of memory
+Peers 3, using 39 KiB of memory
+Peer groups 1, using 64 bytes of memory
+
+Neighbor                  V         AS   MsgRcvd   MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd   PfxSnt Desc
+kind-control-plane(eth00) 4      65000       510       510        3    0    0 00:24:43            1        1 k8s-node
+kind-worker(eth01)        4      65000       510       510        3    0    0 00:24:43            1        1 k8s-node
+kind-worker2(eth02)       4      65000       510       510        3    0    0 00:24:43            1        1 k8s-node
+
+
+> k -n metallb-system exec -it -c frr frr-k8s-daemon-kzpz2 -- vtysh -c 'show bgp summary'
+
+IPv4 Unicast Summary (VRF default):
+BGP router identifier 172.19.0.3, local AS number 65000 vrf-id 0
+BGP table version 2
+RIB entries 3, using 288 bytes of memory
+Peers 1, using 13 KiB of memory
+Peer groups 1, using 64 bytes of memory
+
+Neighbor                  V         AS   MsgRcvd   MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd   PfxSnt Desc
+unnumbered-p2p-peer(net0) 4      65004       512       512        2    0    0 00:25:15            1        2 TOR
+
+Total number of neighbors 1
+
+IPv6 Unicast Summary (VRF default):
+BGP router identifier 172.19.0.3, local AS number 65000 vrf-id 0
+BGP table version 1
+RIB entries 1, using 96 bytes of memory
+Peers 1, using 13 KiB of memory
+Peer groups 1, using 64 bytes of memory
+
+Neighbor                  V         AS   MsgRcvd   MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd   PfxSnt Desc
+unnumbered-p2p-peer(net0) 4      65004       512       512        1    0    0 00:25:15            0        1 TOR
+
+Total number of neighbors 1
+```
+### Topics
+
+#### Remote-AS
```

Comment:
not really sure how this relates here (it says here we can specify internal/external/asn which fits with any model), is there something special about this in regards to unnumbered?

Reply:
I thought that unnumber BGP referred to both interface without values, and ASN without values, but not be the case.

What only confuses me is that 
[neighbor PEER interface remote-as <internal|external|auto|ASN>](https://docs.frrouting.org/en/latest/bgp.html#clicmd-neighbor-PEER-interface-remote-as-internal-external-auto-ASN)

which  talks about unnumber, but when I added this directive  did not work  as expected. in my lab.

I would suggest that this doc documents the API changes for that also, but I will not work on the implementation.

Reply:
>  would suggest that this doc documents the API changes for that also

I think it would be better to keep this focused on unnumbered, when we get to implementing that feature (either if it's before or after this is implemented) we can see how they work together.

Reply:
sounds good, I will remove that section